telegram/utils/helpers.py

```diff
@@ -139,3 +144,25 @@ def effective_message_type(entity):
             return i
 
     return None
+
+
+def passport_auth_url(bot_id, scope, public_key, payload, callback_url=None):
+    """
+    Creates a passport auth url that asks the users for the specified fields
+
+    Args:
+        bot_id (:obj:`int`): Unique identifier for the bot. Can be extracted from the bot token.
+        scope (List[:obj:`str`]): List the names of fields you want to access.
+        public_key (:obj:`str`): Public key of your bot.
+        payload (:obj:`str`): Bot-specified payload
+        callback_url (:obj:`str`, optional): URL to which the user will be redirected.
+
+    """
+    url = 'tg://resolve?domain=telegrampassport' + \
+          '&bot_id={}'.format(bot_id) + \
+          '&scope={}'.format(quote(scope)) + \
+          '&public_key={}'.format(quote(public_key)) + \
+          '&payload={}'.format(quote(payload))
+    if callback_url:
+        url += '&callback_url={}'.format(escape(callback_url))
```

Comment:
escape is not correct here, because we need url encoding, not html escape. `quote` is the correct function